### PR TITLE
fix(docs): Change year in copyright footer automatically to current year

### PR DESCRIPTION
**Pull request by comawill**
Sphinx operates on environment variable SOURCE_DATE_EPOCH to provide reproducible builds.
This PR intends to change the Makefile to set SOURCE_DATE_EPOCH to the latest git revision

Additionally sphinx will only do this correctly, when there are no spaces between the numbers and dash. [\[src\]](https://github.com/sphinx-doc/sphinx/blob/6ffe7be080943130364bac7633bfc1f4f579ef90/sphinx/config.py#L28)

## Checklist

 - [ ] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

**Review by smatting — APPROVED** [review]

**Comment by comawill** [comment]
Do not merge: worked perfectly locally, but has no `git` in github action

**Comment by smatting** [comment]
@comawill Try adding `git` here https://github.com/wireapp/wire-server/blob/develop/nix/default.nix?plain=1#L28
I think @flokli mentioned before that adding git to nix envs could causes problems. But in this case maybe its fine?

**Comment by flokli** [comment]
This would mean the docs and all derivations that depend on them would not become binary reproducible anymore. Can we just drop the year numbers from the footer entirely?

**Comment by smatting** [comment]
@comawill I guess "date" inside the build returns unix epoch 0. I think the problem is that within a nix build you don't have access to the outside world's time. nix tries hard to make all builds deterministic on the input, so it forbids external inputs such as the time.
@flokli do you have an idea how to do this? pass the date from outside to script somehow?

**Comment by flokli** [comment]
As i wrote above, ideally we don't add a year number there at all. If we really want to, not make it automatic based on the current time, but manually, once per year.

**Comment by comawill** [comment]
@smatting using `date +%s` actually seems to work, but I agree with @flokli that this compromises reproducible builds.

My attempt was to depend on the git history here, but within `nix-build` we don't have the history available anymore.

Using the current year or a manual set one does not make a difference as the problems are caused by sphinx here when `SOURCE_DATE_EPOCH` is set. My proposed workaround is now to ignore `SOURCE_DATE_EPOCH` within sphinx.

**Review by flokli — APPROVED** [review]
We probably want to squash this before merge.

**Comment by flokli** [comment]
Already happened in https://github.com/wireapp/wire-server/pull/3110.

**Pull request merged**
